source/extensions/load_balancing_policies/subset/subset_lb.cc

```diff
@@ -828,12 +828,12 @@ void SubsetLoadBalancer::HostSubsetImpl::update(const HostHashSet& matching_host
   auto excluded_hosts_per_locality =
       original_host_set_.excludedHostsPerLocality().filter({cached_predicate})[0];
 
-  HostSetImpl::updateHosts(HostSetImpl::updateHostsParams(
-                               hosts, hosts_per_locality, healthy_hosts, healthy_hosts_per_locality,
-                               degraded_hosts, degraded_hosts_per_locality, excluded_hosts,
-                               excluded_hosts_per_locality),
-                           determineLocalityWeights(*hosts_per_locality), hosts_added,
-                           hosts_removed, absl::nullopt, absl::nullopt);
+  HostSetImpl::updateHosts(
+      HostSetImpl::updateHostsParams(
+          hosts, hosts_per_locality, healthy_hosts, healthy_hosts_per_locality, degraded_hosts,
+          degraded_hosts_per_locality, excluded_hosts, excluded_hosts_per_locality),
+      determineLocalityWeights(*hosts_per_locality), hosts_added, hosts_removed, absl::nullopt,
```

Comment:
This will fix another related bug:

```suggestion
      determineLocalityWeights(*hosts_per_locality), hosts_added, hosts_removed, original_host_set_.weightedPriorityHealth(),
```

Reply:
Fixed, and added integration tests around this behavior.